lib/macroni/ASTConsumers/Safety/SafetyASTConsumer.cpp

```diff
@@ -11,32 +11,23 @@
 #include <memory>
 
 namespace macroni::safety {
-void SafetyASTConsumer::HandleTranslationUnit(clang::ASTContext &Ctx) {
+void SafetyASTConsumer::HandleTranslationUnit(clang::ASTContext &actx) {
   // Match safe blocks.
 
   clang::ast_matchers::MatchFinder finder;
   macroni::safety::safe_block_condition_collector matcher;
   finder.addMatcher(macroni::safety::safe_block_condition_matcher, &matcher);
-  auto driver =
-      macroni::generate_codegen_driver<SafetyDialect,
-                                       macroni::macroni_meta_generator>(Ctx);
-  finder.matchAST(Ctx);
-
-  // Generate the driver.
-
-  driver->push_visitor(std::make_unique<safety_visitor>(
-      matcher.m_safe_block_conditions, driver->mcontext(),
-      driver->get_codegen_builder(), driver->get_meta_generator(),
-      driver->get_symbol_generator(),
-      vast::cg::visitor_view(driver->get_visitor_stack())));
-
-  vast::cg::setup_default_visitor_stack(*driver);
-
-  // Emit the lowered MLIR.
-
-  driver->emit(Ctx.getTranslationUnitDecl());
-  driver->finalize();
-  auto mod = driver->freeze();
+  finder.matchAST(actx);
+
+  auto mctx = vast::mcontext_t();
```

Comment:
As above - you may possibly want to wrap it in your own utility function (to have one point where changes may need to happen).

Reply:
How about if I just create the `mctx` in `macroni::generate_module()` and return it along with the module? That way callers don't have to create the context themselves.